core/src/main/scala/tofu/syntax/foption.scala

```diff
@@ -25,7 +25,7 @@ object foption {
         case x    => x.pure[F]
       }
 
-    def orThrow[E](err: => E)(implicit F: Monad[F], FE: Raise[F, E]): F[A] =
+    def orThrow[E](err: => E)(implicit F: Monad[F], FE: ContravariantRaise[F, E]): F[A] =
```

Comment:
This change makes this ops method **less** universal than it is.

With the current implementation you still can write this:
```scala
  import cats.syntax.all._
  import tofu.syntax.foption._
  def test[F[_] : Monad : Raise[*[_], CommonError]]: F[Unit] = {
    ().some.pure[F].orThrow(ConcreteError())
  }
```
having `Upcast[CommonError, ConcreteError]` in scope. This works due to `raiseUpcast` implicit def. And `ConcreteError` **doesn't need** to be a subclass of `CommonError`. 

If you change the constraint to `ContravariantRaise`, you'll get this code working with direct subclasses via contravariance on E, but lose implicit derivation chain via `raiseUpcast` and the code above will break if `ConcreteError` isn't a subclass of `CommonError` (which is a rare case though).

If you need this change, I would insist on adding extra implicit derivations to make the code above work.